### ISSUE TITLE
Legion: Assertion `future_handles != NULL' failed

### ISSUE THREAD

**Issue by syamajala** [StanfordLegion/legion#1505]
I'm trying @elliottslaughter's fix in this merge request: https://gitlab.com/StanfordLegion/legion/-/merge_requests/832 with S3D and seeing the following error:
```
s3d.x: /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/legion/legion_tasks.cc:11784: void Legion::Internal::SliceTask::handle_future_size(size_t, const Legion::DomainPoint&, std::set<Legion::Internal::RtEvent>&): Assertion `future_handles != NULL' failed.
```

Here is a stack trace:
```
#0  0x00001555528f4238 in nanosleep () from /lib64/libc.so.6
#1  0x00001555528f413e in sleep () from /lib64/libc.so.6
#2  0x0000155548fce512 in Realm::realm_freeze (signal=6) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/realm/runtime_impl.cc:187
#3  <signal handler called>
#4  0x000015555286370f in raise () from /lib64/libc.so.6
#5  0x000015555284db25 in abort () from /lib64/libc.so.6
#6  0x000015555284d9f9 in __assert_fail_base.cold.0 () from /lib64/libc.so.6
#7  0x000015555285bcc6 in __assert_fail () from /lib64/libc.so.6
#8  0x000015554be1a99e in Legion::Internal::SliceTask::handle_future_size (this=0x154bcf000060, future_size=2048, point=..., applied_conditions=...) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/legion/legion_tasks.cc:11784
#9  0x000015554be086d7 in Legion::Internal::PointTask::handle_future_size (this=0x154bfc0240b0, return_type_size=2048, has_return_type_size=true, applied_events=...) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/legion/legion_tasks.cc:7211
#10 0x000015554bdf542e in Legion::Internal::SingleTask::finalize_map_task_output (this=0x154bfc0240b0, input=..., output=..., must_epoch_owner=0x0) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/legion/legion_tasks.cc:2943
#11 0x000015554bdf8bf4 in Legion::Internal::SingleTask::invoke_mapper (this=0x154bfc0240b0, must_epoch_owner=0x0) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/legion/legion_tasks.cc:3580
#12 0x000015554bdfb082 in Legion::Internal::SingleTask::map_all_regions (this=0x154bfc0240b0, must_epoch_op=0x0, defer_args=0x154bff596e20) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/legion/legion_tasks.cc:3983
#13 0x000015554be083c8 in Legion::Internal::PointTask::perform_mapping (this=0x154bfc0240b0, must_epoch_owner=0x0, args=0x154bff596e20) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/legion/legion_tasks.cc:7151
#14 0x000015554c07fc65 in Legion::Internal::Runtime::legion_runtime_task (args=0x154bff596e20, arglen=52, userdata=0xaa0b540, userlen=8, p=...) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/legion/runtime.cc:32304
#15 0x0000155548fa7e30 in Realm::LocalTaskProcessor::execute_task (this=0xa985430, func_id=4, task_args=...) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/realm/proc_impl.cc:1141
#16 0x0000155549019598 in Realm::Task::execute_on_processor (this=0x154bff596ca0, p=...) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/realm/tasks.cc:303
#17 0x000015554901e128 in Realm::UserThreadTaskScheduler::execute_task (this=0x4fe0030, task=0x154bff596ca0) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/realm/tasks.cc:1633
#18 0x000015554901c1af in Realm::ThreadedTaskScheduler::scheduler_loop (this=0x4fe0030) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/realm/tasks.cc:1106
#19 0x0000155549023bbe in Realm::Thread::thread_entry_wrapper<Realm::ThreadedTaskScheduler, &Realm::ThreadedTaskScheduler::scheduler_loop> (obj=0x4fe0030) at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/realm/threads.inl:97
#20 0x0000155549031465 in Realm::UserThread::uthread_entry () at /lustre/scratch/vsyamaj/legion_s3d_nscbc/legion/runtime/realm/threads.cc:1355
#21 0x00001555528792e0 in ?? () from /lib64/libc.so.6
#22 0x0000000000000000 in ?? ()
```

**Comment by lightsighter** [comment]
That's because @elliottslaughter didn't bother to test his changes with control replication first.

**Comment by lightsighter** [comment]
Pull and try again

**Comment by syamajala** [comment]
It works now. I need to do some runs at scale to be sure, but based on what I see so far I don't think this fixes the issue in #1474.
